src/main/java/billgates/view/gui/ActionPanel.java

```diff
@@ -223,4 +213,32 @@ private void deleteEntry() {
     public JButton getDeleteEntryButton() {
         return deleteEntryButton;
     }
+
+    @Override
+    public void view(UserJoinViewModel viewModel) {
+        // If the user join successfully,
+        if (viewModel.isJoined()) {
+            // disable the signInButton, and enable the signOutButton and addEntryButton
+            this.signInButton.setEnabled(false);
+            this.signOutButton.setEnabled(true);
+            this.addEntryButton.setEnabled(true);
+
+            // the usernameField and passwordField shouldn't be editable
+            this.usernameField.setEditable(false);
+            this.passwordField.setEditable(false);
+
+            // enable importMenu
+            TopMenuBar tmb = (TopMenuBar) this.getRootPane().getJMenuBar();
```

Comment:
Remember that you add the `MainFrame` as a instance variable. 
You can utilize that instance variable to complete this task.